pom.xml

```diff
@@ -250,6 +337,41 @@
     </build>
 
     <profiles>
+
+        <!--
+          ~ Profile to run tests using JRE 8.
+          ~
+          ~ It activates if a toolchains configuration file is present.
+          ~ See: https://maven.apache.org/guides/mini/guide-using-toolchains.html
+          -->
+        <profile>
+            <id>java8-tests</id>
+            <activation>
+                <file>
+                    <exists>${user.home}/.m2/toolchains.xml</exists>
+                </file>
+            </activation>
```

Comment:
I do not understand this profile. If the build requires Java 17, how could this be activated and work? What am I missing?

Reply:
The purpose of this profile is to run tests using JRE 8.
Since this can be only done through [Maven Toolchains](https://maven.apache.org/guides/mini/guide-using-toolchains.html), which might be hard to configure for single-time contributors, the profile only activates if a toolchains configuration is present.

Note that the CI always has a toolchains configuration.

Reply:
I hadn't really used it either, but since GitHub automatically sets up the file, it's pretty painless.

I think this pull request makes it so that it's not required (presumably, the CI will test Java 8 if the developer does not), but if you're interested, it has a mojo to automatically try to generate the config file `mvn toolchains:generate-jdk-toolchains-xml`.

Reply:
Cool! :100:
It discovered all 8 JDKs installed on my system.